dev-packages/application-manager/src/generator/frontend-generator.ts

```diff
@@ -183,19 +183,36 @@ if (isMaster) {
                 shell.openExternal(url);
             });
 
-            const saveState = () => {
-                const bounds = newWindow.getBounds();
-                electronStore.set(windowStateName, {
-                    isMaximized: newWindow.isMaximized(),
-                    width: bounds.width,
-                    height: bounds.height,
-                    x: bounds.x,
-                    y: bounds.y
-                })
-            }
-            newWindow.on('close', saveState);
-            newWindow.on('resize', saveState);
-            newWindow.on('move', saveState);
+            // Save the window geometry state on every change
+            const saveWindowState = () => {
+                try {
+                    let bounds;
+                    if (newWindow.isMaximized()) {
+                        bounds = electronStore.get(WINDOW_STATE, {});
+                    } else {
+                        bounds = newWindow.getBounds();
+                    }
+                    electronStore.set(WINDOW_STATE, {
+                        isMaximized: newWindow.isMaximized(),
+                        width: bounds.width,
+                        height: bounds.height,
+                        x: bounds.x,
+                        y: bounds.y
+                    });
+                } catch (e) {
+                    console.error("Error while saving window state.", e);
+                }
+            };
+            let delayedSaveTimeout;
+            const saveWindowStateDelayed = () => {
```

Comment:
Nit: I think the correct term for "Delayed" is ["Debounced"](https://davidwalsh.name/javascript-debounce-function). If you agree, you could replace all occurrences of "Delayed" with "Debounced".

Reply:
I think both terms are valid...